### DISCUSSION TITLE
Setting styles in templated control causes control to crash under windows

### DISCUSSION THREAD

**Discussion by kucint** [unoplatform/uno.extensions#2316] [Q&A] [answered]
Create a simple control and try to set its style to some default style:

```cs
internal class MyControl : Control
{
    public MyControl()
    {
        DefaultStyleKey = typeof(MyControl);
        Style = _defaultStyle;  // crashes with COMException
    }
}
```
This causes exception: 
_COMException: 'No installed components were detected.
Cannot apply a Style with TargetType 'Controls.MyControl'_

the app crashes only under net8.0-windows10.0.19041
under net8.0-desktop all is OK

to reproduce: 
- take the app [UnoAppTemplateBindingWorking.zip](https://github.com/unoplatform/uno.extensions/files/15241569/UnoAppTemplateBindingWorking.zip) written by @dr1rrb for the purpose of [#2270](https://github.com/unoplatform/uno.extensions/discussions/2270) 
- Add `net8.0-windows10.0.19041` to its `TargetFrameworks`. 
- Run as `net8.0-windows10.0.19041` 'unpackaged'
- It crashes (note that if ran under net8.0-desktop, the app works fine)

Alternatively take this one to reproduce: [UnoNoInstalledComponentsApp.zip](https://github.com/user-attachments/files/15534406/UnoNoInstalledComponentsApp.zip)

Tested only with windows:
` "Uno.Sdk": "5.2.139"`

**Comment by MartinZikmund** [comment, marked as answer]
@kucint Apparently there is some limitiation in WinUI that requires controls to have default styles defined. To do so, you can create a XAML resource dictionary like:

```xml
<ResourceDictionary
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:my="using:UnoAppTemplateBindingNotWorking.Controls">
    <Style TargetType="my:MyControl" />
</ResourceDictionary>
```

And then reference that dictionary in `App.xaml`:

```xml
<Application
    x:Class="UnoAppTemplateBindingNotWorking.App"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
    <Application.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <ResourceDictionary Source="ms-appx:///MyDictionary.xaml" />
            </ResourceDictionary.MergedDictionaries>
        </ResourceDictionary>
    </Application.Resources>
</Application>
```
